plugins/SpectrumAnalyzer/SaControls.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * SaControls.h - declaration of SaControls class.
+ *
+ * Copyright (c) 2019 Martin Pavelek <he29/dot/HS/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef SACONTROLS_H
+#define SACONTROLS_H
+
+#include <QPainter>
+
+#include "ComboBoxModel.h"
+#include "EffectControls.h"
+
+//#define SA_DEBUG 1	// define SA_DEBUG to enable performance measurements
+
+// Frequency ranges (in Hz).
+// Full range is defined by LOWEST_LOG_FREQ and current sample rate.
+const int LOWEST_LOG_FREQ = 10;		// arbitrary low limit for log. scale, >1
+
+enum FREQUENCY_RANGES
+{
+	FRANGE_FULL = 0,
+	FRANGE_AUDIBLE,
+	FRANGE_BASS,
+	FRANGE_MIDS,
+	FRANGE_HIGH
+};
+
+const int FRANGE_AUDIBLE_START = 20;
```

Comment:
I originally meant these to be moved in fft_helpers, to be shared by other plugins that use FFT, but I wasn't sure it is the right place for it, so I just kept it here for now. I imagine the EQ plugin may use it as well (switchable ranges are among the requested features I saw in one issue) and it may be useful to have some sort of "global" conventions for ranges -- what is considered audible range, what is bass, what loudness is considered inaudible etc.. That may require a new file outside the plugin folder, though, and that did not seem to me like a good idea since I'm completely new to the project. :))

So, I'm not sure what to do about these constants either. But I agree that the current placement probably isn't optimal.

Reply:
OK, in this case, it makes sense. You can leave them here for the time being.